### PR TITLE
[Webhook] Added component documentation for use in combination with Mailer

### PR DESCRIPTION
**Pull request by TimoBakx**
Replaces
* #17964 

#SymfonyHackday

### CODE REVIEW

**Review by dsdeboer — APPROVED** [review]
👏

**Comment by OskarStark** [comment]
A personal big big thank you for driving this forward. Thank you Timo 🙏

**Comment by OskarStark** [comment]
Please rebase on 6.3, thanks

**Comment by alexandre-daubois** [comment]
Congrats on this new version 🥳

**Comment by TimoBakx** [comment]
@OskarStark thank you for reminding me.

- Rebased and retargeted to 6.3
- Squashed commits into 1
- Fixed inconsistent internal naming in a literal string
- Added `type` attribute in XML config (see https://github.com/symfony/symfony/pull/52978)

The code block check fails due to Webhook not being installed in the test application ([symfony-tools/symfony-application](https://github.com/symfony-tools/symfony-application.git)). Should I add it there?

File: webhook.rst [hunk 1]
```diff
@@ -0,0 +1,133 @@
+Webhook
+=======
+
+.. versionadded:: 6.3
+
+    The Webhook component was introduced in Symfony 6.3
+
+The webhook component is used to respond to remote webhooks to trigger actions
```
Comment (MrYamous) [inline webhook.rst]:
```suggestion
The Webhook component is used to respond to remote webhooks to trigger actions
```

File: webhook.rst [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+in your application. This document focusses on using webhooks to listen to remote
```
Comment (MrYamous) [inline webhook.rst]:
```suggestion
in your application. This document focuses on using webhooks to listen to remote
```

File: reference/configuration/framework.rst [hunk 3]
```diff
@@ -3625,6 +3625,16 @@ enabled
...
 
+webhook
+~~~~~~~
+
+.. versionadded:: 6.3
+
+    The Webhook configuration was introduced in Symfony 6.3.
+
+The ``webhook`` option (and its children) are used to configure the webhooks
+defined in your application. Read more about the options in the :ref:`Webhooks documentation <webhook>`.
```
Comment (MrYamous) [inline reference/configuration/framework.rst]:
```suggestion
defined in your application. Read more about the options in the :ref:`Webhook documentation <webhook>`.
```

File: webhook.rst [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+------------
+
+.. code-block:: terminal
+
+    $ composer require symfony/webhook
+
+Usage in combination with the Mailer component
+----------------------------------------------
+
+When using a third-party mailer, you can use the webhook component to receive
```
Comment (MrYamous) [inline webhook.rst]:
```suggestion
When using a third-party mailer, you can use the Webhook component to receive
```

File: webhook.rst [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+webhook calls from the third-party mailer.
+
+In this example Mailgun is used with ``'mailer_mailgun'`` as webhook type.
+Any type name can be used as long as it's unique. Make sure to use it in the
+routing configuration, the webhook URL and the Remote Event Consumer.
+
+Install the third party mailer as described in the documentation of the mailer
+component.
+
+The webhook component routing needs to be defined:
```
Comment (MrYamous) [inline webhook.rst]:
```suggestion
The Webhook component routing needs to be defined:
```

File: webhook.rst [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+        };
+
+Currently, the following third-party mailer services support webhooks:
+
+=============== ==========================================
+Mailer service  Parser service name
+=============== ==========================================
+Mailgun         ``mailer.webhook.request_parser.mailgun``
+Postmark        ``mailer.webhook.request_parser.postmark``
+Sendgrid        ``mailer.webhook.request_parser.sendgrid``
```
Comment (MrYamous) [inline webhook.rst]: IIIC Sendgrid was added in 6.4

File: webhook.rst [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+path behind the domain you're using.
+
+Mailgun will provide a secret for the webhook. Add this secret to your ``.env``
+file:
+
+.. code-block:: env
+
+    MAILER_MAILGUN_SECRET=your_secret
+
+With this done, you can now add Remote Event Consumers to react to the webhooks::
```
Comment (MrYamous) [inline webhook.rst]:
```suggestion
With this done, you can now add RemoteEvent Consumer to react to the webhooks::
```
Not sure about space and/or capitalize for consumer

**Comment by MrYamous** [comment]
I am glad to see some news for this component documentation, congratulations

File: webhook.rst (hunk 6 above)
Comment (OskarStark) [reply webhook.rst]: If this is true we should only list 6.3 ones. And add Sendgrid in 6.4 after merge with a versionadded directive

File: webhook.rst (hunk 7 above)
Comment (OskarStark) [reply webhook.rst]: a RemoteEvent consumer

File: webhook.rst (hunk 6 above)
Comment (TimoBakx) [reply webhook.rst]: You're right. 6.3 only has Mailgun and Postmark.
6.4 added Sendgrid.

I removed Sendgrid for now. I can add it in a separate PR for 6.4.

**Comment by TimoBakx** [comment]
Thank you @MrYamous and @OskarStark for the feedback. I've updated everything accordingly.

File: reference/attributes.rst [hunk 8]
```diff
@@ -112,6 +112,11 @@ Each validation constraint comes with a PHP attribute. See
 
 * :doc:`HasNamedArguments </validation/custom_constraint>`
 
+Webhook
```
Comment (MrYamous) [inline reference/attributes.rst]:
```suggestion
RemoteEvent
```

RemoteEvent is required by Webhook but i think it's appropriate to mention here the component providing this attribute

File: webhook.rst [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+
+Usage in combination with the Mailer component
+----------------------------------------------
+
+When using a third-party mailer, you can use the Webhook component to receive
+webhook calls from the third-party mailer.
+
+In this example Mailgun is used with ``'mailer_mailgun'`` as webhook type.
+Any type name can be used as long as it's unique. Make sure to use it in the
+routing configuration, the webhook URL and the Remote Event Consumer.
```
Comment (MrYamous) [inline webhook.rst]:
```suggestion
routing configuration, the webhook URL and the RemoteEvent consumer.
```

File: webhook.rst [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+
+Install the third party mailer as described in the documentation of the mailer
```
Comment (MrYamous) [inline webhook.rst]: We may can add a link to Mailer documentation
Edit : And in Mailer documentation, maybe mention this new page too :)

Comment (TimoBakx) [reply webhook.rst]: Indeed. I also want to link from the Mailer documentation to the webhook as well.

Comment (TimoBakx) [reply webhook.rst]: Can I link to the third party bit directly? If so, how?

Comment (OskarStark) [reply webhook.rst]: As we don't have much info for a third party bridge in the docs we could link to the overview or link to the README in the code repo.

What would you prefer? 

Btw, WDYT about this PR? https://github.com/symfony/symfony-docs/pull/19263

Comment (TimoBakx) [reply webhook.rst]: That PR would be a good addition, @OskarStark. Nice.

As for the link, see my comment below. I'm not sure if that works as I hope it works. 😅

File: webhook.rst [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+
+When using a third-party mailer, you can use the Webhook component to receive
+webhook calls from the third-party mailer.
+
+In this example Mailgun is used with ``'mailer_mailgun'`` as webhook type.
+Any type name can be used as long as it's unique. Make sure to use it in the
+routing configuration, the webhook URL and the RemoteEvent consumer.
+
+Install the third party mailer as described in the documentation of the
+:doc:`Mailer component </mailer>`.
```
Comment (TimoBakx) [inline webhook.rst]:
```suggestion
:doc:`Mailer component </mailer_using-a-3rd-party-transport>`.
```

@OskarStark would this notation work to link directly to that part of the documentation?

Comment (MrYamous) [reply webhook.rst]: You should add a target on Mailer documentation page right before the title of the section like `.. _mailer-using-3rd-party-transport:` as it's done for example with Transport Setup on top of the page
Then you can use this in a ref directive

See [documentation format](https://symfony.com/doc/current/contributing/documentation/format.html#adding-links) for more infos

Comment (TimoBakx) [reply webhook.rst]: Thanks! Updated accordingly.

**Comment by OskarStark** [comment]
Can you please rebase? I fixed the Build error

**Comment by TimoBakx** [comment]
@OskarStark Rebased & squashed.

**Review by OskarStark — APPROVED** [review]

**Comment by OskarStark** [comment]
Big big thank you!

**Comment by OskarStark** [comment]
Maybe someone can provide more infos about which bridges support webhooks in mailer.rst starting from 6.4 branch

**Comment by alexandre-daubois** [comment]
Congrats! :smiley: 

I'm looking at bridges :+1:

**Comment by TimoBakx** [comment]
> Maybe someone can provide more infos about which bridges support webhooks in mailer.rst starting from 6.4 branch

Working in the PRs for those. :)

**Comment by OskarStark** [comment]
Only one please :-D

**Comment by TimoBakx** [comment]
> Only one please :-D

Sorry, I was too fast. >_<

**Pull request merged**